services/src/api/model/datatypes.rs

```diff
@@ -1032,3 +1296,267 @@ impl TryFrom<SerializablePalette> for Palette {
+                        if c == '%' {
+                            state = FormatStrLoopState::Percent("%".to_string());
+                        }
+                    }
+                    FormatStrLoopState::Percent(ref mut s) => {
+                        s.push(c);
+
+                        if c == '%' {
+                            // was escaped percentage
+                            state = FormatStrLoopState::Normal;
+                        } else if c.is_ascii_alphabetic() {
+                            if has_tz_values.contains(s.as_str()) {
+                                has_tz = true;
+                            }
+                            if has_time_values.contains(s.as_str()) {
+                                has_time = true;
+                            }
+
+                            state = FormatStrLoopState::Normal;
+                        }
+                    }
+                }
+            }
+
+            (has_tz, has_time)
+        };
+
+        DateTimeParseFormat {
+            fmt,
+            has_tz,
+            has_time,
+        }
+    }
+
+    pub fn unix() -> Self {
+        let fmt = "%s".to_owned();
+        Self {
+            fmt,
+            has_tz: false,
+            has_time: true,
+        }
+    }
+
+    pub fn ymd() -> Self {
+        let fmt = "%Y-%m-%d".to_owned();
+        Self {
+            fmt,
+            has_tz: false,
+            has_time: false,
+        }
+    }
+
+    pub fn is_empty(&self) -> bool {
+        self.fmt.is_empty()
+    }
+
+    fn _to_parse_format(&self) -> &str {
+        &self.fmt
+    }
+}
+
+enum FormatStrLoopState {
+    Normal,
+    Percent(String),
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub struct NoGeometry;
+
+impl From<geoengine_datatypes::primitives::NoGeometry> for NoGeometry {
+    fn from(_: geoengine_datatypes::primitives::NoGeometry) -> Self {
+        Self {}
+    }
+}
+
+impl From<NoGeometry> for geoengine_datatypes::primitives::NoGeometry {
+    fn from(_: NoGeometry) -> Self {
+        Self {}
+    }
+}
+
+impl ToSchema for NoGeometry {
+    fn schema() -> utoipa::openapi::Schema {
+        use utoipa::openapi::*;
+        ObjectBuilder::new()
+            .default(Some(serde_json::Value::Null))
+            .example(Some(serde_json::Value::Null))
+            .into()
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize, ToSchema)]
+pub struct MultiPoint {
+    coordinates: Vec<Coordinate2D>,
+}
+
+impl From<geoengine_datatypes::primitives::MultiPoint> for MultiPoint {
+    fn from(value: geoengine_datatypes::primitives::MultiPoint) -> Self {
+        Self {
+            coordinates: value.points().iter().map(|x| (*x).into()).collect(),
+        }
+    }
+}
+
+impl From<MultiPoint> for geoengine_datatypes::primitives::MultiPoint {
+    fn from(value: MultiPoint) -> Self {
+        Self::new(value.coordinates.iter().map(|x| (*x).into()).collect()).unwrap()
```

Comment:
use `into_iter` instead of `iter`

Reply:
[44637dd](https://github.com/geo-engine/geoengine/pull/662/commits/44637ddbac75b8e5ddfd9ab6d09f7776eb6cbdab)